src/extension/wps/src/main/java/au/org/emii/gogoduck/worker/CSVConverter.java

```diff
@@ -0,0 +1,257 @@
+    private List<String> lines = new ArrayList<String>();
+    private List<String> metadata = new ArrayList<String>();
+    private List<String> attributeMetadata = new ArrayList<String>();
+    private List<Map> attributes = new ArrayList<Map>();
+    private StringBuilder csvString = new StringBuilder();
+
+    private final String NEW_LINE = System.getProperty("line.separator");
+    private final String COMMENT = "# ";
+    private static final Logger logger = LoggerFactory.getLogger(CSVConverter.class);
+
+    public CSVConverter() {}
+
+    @Override
+    public void init() {
+        this.ncksPath = GoGoDuckConfig.ncksPath;
+        this.ncpdqPath = GoGoDuckConfig.ncpdqPath;
+    }
+
+    @Override
+    public String getMimeType() { return mimeType; }
+
+    @Override
+    public String getExtension() { return extension; }
+
+    @Override
+    public Path convert(Path outputFile) throws GoGoDuckException {
+        String baseFilename = FilenameUtils.getBaseName(String.valueOf(outputFile));
+        try {
+            File unpackedNetCDF = unpackNetCDF(outputFile.toFile());
+            Path csvPath = executeNcksAndOutputToFile(baseFilename, unpackedNetCDF.toPath());
+            Files.delete(unpackedNetCDF.toPath());
+            return csvPath;
+        } catch (Exception e) {
+            throw new GoGoDuckException(String.format("Could not convert output to csv: '%s'", e.getMessage()));
+        }
+    }
+
+    private File unpackNetCDF(File netCDFInputFile) throws Exception {
+        List<String> ncpdqCommand = new ArrayList<>();
+        File unpackedNetCDF = File.createTempFile("tmp", ".nc");
+
+        ncpdqCommand.add(ncpdqPath);
+        ncpdqCommand.add("-U");
+        ncpdqCommand.add("-O");
+        ncpdqCommand.add(netCDFInputFile.toString());
+        ncpdqCommand.add(unpackedNetCDF.getPath());
+
+        GoGoDuck.execute(ncpdqCommand);
+        return unpackedNetCDF;
+    }
+
+    private Path executeNcksAndOutputToFile(String baseFilename, Path tempFile) throws IOException, InterruptedException {
+        List<String> ncksCommand = new ArrayList<>();
+        ncksCommand.add(ncksPath);
+        ncksCommand.add("--no_nm_prn");
+        ncksCommand.add("-a");
+        ncksCommand.add(tempFile.toString());
+
+        logger.info(ncksCommand.toString());
+        
+        ProcessBuilder processor = new ProcessBuilder(ncksCommand);
+        processor.redirectError(ProcessBuilder.Redirect.PIPE);
+
+        final Process process = processor.start();
+        InputStream is = process.getInputStream();
+        InputStreamReader isr = new InputStreamReader(is);
+        BufferedReader br = new BufferedReader(isr);
+
+        InputStreamReader errorStream = new InputStreamReader(process.getErrorStream());
+        BufferedReader bufferedErrorStream = new BufferedReader(errorStream);
+        
+        StringBuilder builder = new StringBuilder();
+
+        String line;
+        while ((line = br.readLine()) != null) {
```

Comment:
Yep, you're right, because `processor.start()` will actually start the process. Still, consuming stderr should be done regardless of catching exceptions and probably just here, below the consumption of stdout.